protocol/src/main/scala/io/mediachain/protocol/CborSerialization.scala

```diff
@@ -693,8 +693,8 @@ object CborSerialization {
 
     override def fromCValue(cValue: CValue): Xor[DeserializationError, MultihashReference] = {
       cValue match {
-        case CTaggedValue(tag, CBytes(value)) if tag == MultihashReference.cborLinkTag =>
-          MultiHash.fromBytes(value)
+        case CTaggedValue(tag, CString(value)) if tag == MultihashReference.cborLinkTag =>
+          MultiHash.fromBase58(value)
```

Comment:
Yeah, I can't tell whether this is a design decision or just an implementation bug 😄   The current js ipld implementation assumes that byte strings are multiaddrs, so if you give it a multihash instead it will fail to decode.  But it parses base58 multihash strings just fine and resolves them as ipfs links.